api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -893,4 +903,74 @@ public CohortDefinition getNumberOfPatientsWhoAbandonedArtDuringPreviousMonthFor
             hivMetadata.getArtDatePickup().getConceptId()));
     return cd;
   }
+
+  /**
+   * Get all patients enrolled in PRE-ART program id 1, with date enrolled less than startDate
+   *
+   * @return CohortDefinition
+   */
+  private CohortDefinition
+      getAllPatientsEnrolledInPreArtProgramWithDateEnrolledLessThanStartDateA1() {
+    SqlCohortDefinition sqlPatientsEnrolledInPreART = new SqlCohortDefinition();
+    sqlPatientsEnrolledInPreART.setName("patientsEnrolledInPreART");
+    sqlPatientsEnrolledInPreART.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    sqlPatientsEnrolledInPreART.addParameter(new Parameter("endDate", "End Date", Date.class));
+    sqlPatientsEnrolledInPreART.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientsEnrolledInPreART.setQuery(
+        ResumoMensalQueries.getAllPatientsEnrolledInPreArtProgramWithDateEnrolledLessThanStartDate(
+            hivMetadata.getHIVCareProgram().getProgramId()));
+
+    return sqlPatientsEnrolledInPreART;
+  }
+
+  /**
+   * Get all patients registered in encounterType 5 or 7, with date enrolled less than startDate
+   *
+   * @return CohortDefinition
+   */
+  private CohortDefinition
+      getAllPatientsRegisteredInEncounterType5or7WithEncounterDatetimeLessThanStartDateA1() {
+    SqlCohortDefinition sqlPatientsRegisteredEncounterType5or7 = new SqlCohortDefinition();
```

Comment:
We try to use as less SQL as possible. This cohort should be implemented using one of the built-ins, in this case `EncounterCohortDefinition`.

Reply:
```
EncouterCohortDefinition cd = new EncounterCohortDefinition();
cd.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
cd.addParameter(new Parameter("locationList", "Location", Location.class));
cd.addEncounterType(hivMetadata.getARVAdultInitialEncounterType());
cd.addEncounterType(hivMetadata.getARVPediatriaInitialEncounterType());
return cd;
```